### DISCUSSION TITLE
MSA 환경에서의 메시징 시스템 (Apache Kafka vs RabbitMQ)

### DISCUSSION THREAD

**Discussion by JiHongKim98** [100-hours-a-week/discussions#27] [Deepdive]
## 메시지 브로커를 딥다이브 주제로 선택한 이유

- 2차 스프린트 이후 DAO 프로젝트를 모놀리틱에서 `MSA` 로 마이그레이션 예정
   - 사유 : node.js 기반의 애플리케이션 추가 구현 예정
   - 마이크로 서비스들 간의 도메인 이벤트 발행 및 데이터베이스 SYNC 를 맞추기 위해 Kafka 도입 예정

- Kafka 와 RabbitMQ
   - 대중적으로 가장 많이 쓰이는 2가지 기술 스택이므로 선택
      - 많은 레퍼런스 및 적용 사례 참고 가능
   - Redis PUB/SUB 은 고려하지 않음
      - 내구성의 한계 및 `at least once` 등을 고려하기 어렵기 때문에 고려하지 않음

**Comment by startupcodekr** [comment]
1. 제목 변경해주세요
MSA 환경에서의 메시징 시스템 선택: Apache Kafka vs RabbitMQ 비교 분석 및 DAO 프로젝트 마이그레이션 전략

2. 1. 왜 msa를 하는지 제목에 추가해주세요.
